### DISCUSSION TITLE
MppSolar - Setting Charging Stage

### DISCUSSION THREAD

**Discussion by brunopiras** [syssi/esphome-pipsolar#74] [General]
Hello there,

there is a chance to insert a command to set Floating mode?

I've a lithium battery pack but the bms it does not communicate with the inverter, I need to set floating mode when the battery is at 100%.

The command :

PCSTNN

NN = 00 (Auto)
NN = 01 (Stage 2)
NN = 02 (Stage 3)

Thanks

otherwise, I can send these with the uart.write ?

**Comment by meccip** [comment]
Hello,
If you can control the bms with an automation, you can stop the charging and the inverter will enter Float mode. For example i use nodered and with data from esp-jk-bms, i can stop charging when a cell rich 3.45v. After some time the inverter will enter in Float Mode and nodered send command to bms to start charging. The inverter will stay in Float at the given float_voltage
